test/SlowTests/Issues/RavenDB-23085.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using FastTests.Voron;
+using Sparrow.Server.Platform;
+using Tests.Infrastructure;
+using Voron;
+using Voron.Impl.Paging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_23085(ITestOutputHelper output) : StorageTest(output)
+{
+    [RavenFact(RavenTestCategory.Voron)]
+    public void CanAccountForLockedMemory()
+    {
+        long locked = 0;
+        EventHandler<long> handler = (s, amount) => locked += amount;
+        // problematic - since this is a global state
+        RequireFileBasedPager();
+        try
+        {
+            MemoryLockUsage.MemoryLockedCalled += handler;
+            
+            (Pager pager, Pager.State state) = Pager.Create(Options,
+                Options.BasePath.Combine("data.locked").FullPath,64*1024, 
+                Pal.OpenFileFlags.LockMemory | Pal.OpenFileFlags.WritableMap);
+            Assert.NotEqual(0, locked);
+            var prev = locked;
+            pager.EnsureContinuous(ref state, 1028, 1);
+            Assert.True(prev < locked);
+            pager.Dispose();
+            state.Dispose();
+            Assert.Equal(0, locked);
```

Comment:
Wouldn't this be introducing false positive failures? `MemoryLockUsage.MemoryLockedCalled` can be called by someone else at the time of test execution. I don't think we're guaranteed that it will be always `0`

Reply:
Correct, that is a good point. Fixed.